nimble/controller/src/ble_ll_sched.c

```diff
@@ -961,20 +966,25 @@ ble_ll_sched_adv_reschedule(struct ble_ll_sched_item *sch, uint32_t *start,
              * with original duration.
              */
             before = NULL;
-            orig_start = sch->start_time;
+            sch->start_time = orig_start;
             entry = start_overlap;
             sch->end_time = sch->start_time + duration;
             while (1) {
                 next_sch = entry->link.tqe_next;
                 if ((int32_t)(sch->end_time - entry->start_time) <= 0) {
-                    rand_ticks = entry->start_time - sch->end_time;
-                    before = entry;
-                    TAILQ_INSERT_BEFORE(before, sch, link);
-                    break;
-                } else {
-                    sch->start_time = entry->end_time;
-                    sch->end_time = sch->start_time + duration;
```

Comment:
here we can indeed move `sch->start_time` beyond `orig_start_time + max_delay_ticks` but this can only happen for last overlapping event which is handled in the condition below.

Reply:
but there is issue in the handle for the below condition.
The sch may overlap next_sch if next_sch is not NULL **if (sch->end_time >next_sch->start_time)**
```
                    } else {
                        if (next_sch == NULL) {
                            TAILQ_INSERT_TAIL(&g_ble_ll_sched_q, sch, link);
                        } else {
                            TAILQ_INSERT_BEFORE(next_sch, sch, link);
                        }
                    }
```

Reply:
I think
`rand_ticks = (orig_start + max_delay_ticks) - sch->start_time;  ` 
equal to 
`rand_ticks = (orig_end + max_delay_ticks) - sch->end_time;  `

If we can insert sch into link, this means  `sch->end_time <= orig_end + max_delay_ticks`, so we can make sure sch will not overflow(max_delay). If `sch` overlaps `next_sch`, `end_overlap` should be `next_sch` (sch->end_time <= orig_end + max_delay < end_overlap->next_sch->start_time).

Reply:
`sch->end_time > next_sch->start_time` is impossible, because `next_sch` is the next element after `entry` (see line 968). `entry` is the last element overlapping `sch` so `next_sch` cannot overlap `sch`.

Reply:
understood.